simtools/simtel/simtel_events.py

```diff
@@ -1,194 +1,253 @@
         '''
-        Plot all histograms into pdf pages and save the figure as a pdf file.
+        Count number of triggered events within a certain energy range and core radius.
 
         Parameters
         ----------
-        combinedHists: list
-            Hisograms from all files combined by the function _combinedHistogramFiles
-        figName: str
-            Name of the output figure file.
+        energyRange: Tuple (len 2)
+            Max and min energy of energy range, e.g. energyRange=(100 * u.GeV, 10 * u.TeV)
+        coreMax: astropy.Quantity (distance)
+            Maximum core radius for selecting showers, e.g. coreMax=1000 * u.m
+
+        Returns
+        -------
+        int
+            Number of triggered events.
         '''
+        energyRange = self._validateEnergyRange(energyRange)
+        coreMax = self._validateCoreMax(coreMax)
+
+        isInEnergyRange = list(map(
+            lambda e: e > energyRange[0] and e < energyRange[1],
+            self.summaryEvents['energy']
+        ))
+        isInCoreRange = list(map(
+            lambda r: r < coreMax,
+            self.summaryEvents['r_core']
+        ))
+        return np.sum(np.array(isInEnergyRange) * np.array(isInCoreRange))
+
+    @u.quantity_input(coreMax=u.m)
+    def selectEvents(self, energyRange=None, coreMax=None):
+        '''
+        Select sim_telarray events within a certain energy range and core radius.
 
-        def _get_bins(hist, axis=0):
-            ax_str = 'x' if axis == 0 else 'y'
-            return np.linspace(
-                hist['lower_' + ax_str],
-                hist['upper_' + ax_str],
-                hist['n_bins_' + ax_str] + 1
-            )
-
-        def _get_ax_lim(hist, axis=0):
-            if np.sum(hist['data']) == 0:
-                return 0, 1
-
-            bins = _get_bins(hist, axis=axis)
-
-            if hist['data'].ndim == 1:
-                non_zero = np.where(hist['data'] != 0)
-            else:
-                marginal = np.sum(hist['data'], axis=axis)
-                non_zero = np.where(marginal != 0)
+        Parameters
+        ----------
+        energyRange: Tuple (len 2)
+            Max and min energy of energy range, e.g. energyRange=(100 * u.GeV, 10 * u.TeV)
+        coreMax: astropy.Quantity (distance)
+            Maximum core radius for selecting showers, e.g. coreMax=1000 * u.m
+
+        Returns
+        -------
+        list
+            List of events.
+        '''
+        energyRange = self._validateEnergyRange(energyRange)
+        coreMax = self._validateCoreMax(coreMax)
 
-            return bins[non_zero[0][0]], bins[non_zero[0][-1] + 1]
+        selectedEvents = list()
+        for file in self.inputFiles:
+            with SimTelFile(file) as f:
 
-        pdfPages = PdfPages(figName)
-        for hist in combinedHists:
+                for event in f:
+                    energy = event['mc_shower']['energy']
+                    x_core = event['mc_event']['xcore']
```

Comment:
It could be a little more efficient if you do the second calculation after performing the first test (you can then avoid unnecessary calculations). Won't make a big difference though I think.

Reply:
Yep, good catch. I changed it now